src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/AvailabilityResourceIntegrationTest.kt

```diff
@@ -0,0 +1,210 @@
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.response.AvailabilityResponse
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.VideoBookingRepository
+import java.time.LocalDate
+import java.time.LocalTime
+
+class AvailabilityResourceIntegrationTest : IntegrationTestBase() {
+  @Autowired
+  private lateinit var videoBookingRepository: VideoBookingRepository
+
+  @Test
+  fun `should confirm availability for a court booking when the prison room is free`() {
+    val bookingCreator = "BOOKING_CREATOR"
+
+    videoBookingRepository.findAll() hasSize 0
+
+    // With no bookings present, the availability check should succeed
+    val availabilityRequest = AvailabilityRequest(
+      bookingType = BookingType.COURT,
+      courtOrProbationCode = "DRBYMC",
+      prisonCode = WERRINGTON,
+      date = LocalDate.now().plusDays(1),
+      mainAppointment = LocationAndInterval(
+        werringtonLocation.key,
+        Interval(
+          start = LocalTime.of(12, 0),
+          end = LocalTime.of(12, 30),
+        ),
+      ),
+    )
+
+    val availabilityResponse = webTestClient.availabilityCheck(bookingCreator, availabilityRequest)
+
+    assertThat(availabilityResponse).isNotNull
+    with(availabilityResponse) {
+      assertThat(availabilityOk).isTrue()
+      assertThat(alternatives).isEmpty()
+    }
+  }
+
+  @Test
+  fun `should offer alternatives for a court booking when the prison room is already occupied`() {
+    val bookingCreator = "BOOKING_CREATOR"
+
+    videoBookingRepository.findAll() hasSize 0
+
+    prisonSearchApi().stubGetPrisoner("A1111AA", WERRINGTON)
+    locationsInsidePrisonApi().stubPostLocationByKeys(setOf(werringtonLocation.key), WERRINGTON)
+
+    val courtBookingRequest = courtBookingRequest(
+      courtCode = "DRBYMC",
+      prisonerNumber = "A1111AA",
+      prisonCode = WERRINGTON,
+      location = werringtonLocation,
+      startTime = LocalTime.of(12, 0),
+      endTime = LocalTime.of(12, 30),
+      comments = "integration test court booking comments",
+    )
+
+    webTestClient.createBooking(bookingCreator, courtBookingRequest)
+
+    // Do the availability check for a booking at the same time as the existing booking above
+    val availabilityRequest = AvailabilityRequest(
+      bookingType = BookingType.COURT,
+      courtOrProbationCode = "DRBYMC",
+      prisonCode = WERRINGTON,
+      date = LocalDate.now().plusDays(1),
+      mainAppointment = LocationAndInterval(
+        werringtonLocation.key,
+        Interval(
+          start = LocalTime.of(12, 0),
+          end = LocalTime.of(12, 30),
+        ),
+      ),
+    )
+
+    val availabilityResponse = webTestClient.availabilityCheck(bookingCreator, availabilityRequest)
+
+    assertThat(availabilityResponse).isNotNull
+    with(availabilityResponse) {
+      assertThat(availabilityOk).isFalse()
+      assertThat(alternatives).isNotEmpty()
```

Comment:
The `isNotEmpty` check not really needed as checking the size on the following line.

Is it worth asserting on what the alternatives are? as in times for example?

Reply:
Done. Other tests on the returned values are now done in the new AvailablilityServiceTest module.